tsconfig.json

```diff
@@ -0,0 +1,47 @@
+{
+  "compilerOptions": {
+    /* Basic Options */
+
+    // https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping as now we support node14.
+    "lib": ["ES2020"],
+    "module": "commonjs",
+    "target": "ES2020",
+
+    "jsx": "preserve",
+    "declaration": false,
+    "rootDir": "./",
+    "outDir": "dist",
+
+    /* Strict Type-Checking Options */
+    "strict": true,
+    "noImplicitAny": true,
+    "strictNullChecks": true,
+    "strictFunctionTypes": true,
+    "strictBindCallApply": true,
+    "strictPropertyInitialization": true,
+    "noImplicitThis": true,
+    "alwaysStrict": true,
+    "useUnknownInCatchVariables": false,
+
+    /* Additional Checks */
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true,
+
+    /* Module Resolution Options */
+    "moduleResolution": "node",
+    "baseUrl": "./src",
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "allowUmdGlobalAccess": true,
+    "resolveJsonModule": true,
+
+    /* Advanced Options */
+    "skipLibCheck": true /* Skip type checking of declaration files. */,
+    "forceConsistentCasingInFileNames": true /* Disallow inconsistently-cased references to the same file. */
+  },
+  "files": ["./extend-expect.d.ts"],
```

Comment:
Do we still need explicit type defs? We should be able to move them to relevant files.

Reply:
We still will need to maintain a separated type defs file as we need to modify the jest namespace. it's possible to move it on the `src` but i don't see much of a benefit vs having it on the root folder(easier to find).